pkg/kubectl/cmd/resource/get.go

```diff
@@ -190,9 +192,13 @@ func (options *GetOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args
 		options.ExplicitNamespace = false
 	}
 
+	options.IncludeUninitialized = cmdutil.ShouldIncludeUninitialized(cmd, false)
+
 	switch {
 	case options.Watch || options.WatchOnly:
-
+		// include uninitialized objects when watching on a single object
+		// unless explicitly set --include-uninitialized=false
+		options.IncludeUninitialized = cmdutil.ShouldIncludeUninitialized(cmd, len(args) == 2)
```

Comment:
I know this is preexisting, but `len(args) == 2` is not a reliable way to determine single item... `pods/foo` refers to a single item

Reply:
> pods/foo refers to a single item

Yes, right. Seems I did not take this into consideration. Thanks for pointing out. 👍👍
Will update it later.

Reply:
Yeah, the correct way to find single item is to use the "Result" object from the resource builder.